corehq/apps/app_manager/models.py

```diff
@@ -1233,6 +1233,14 @@ class MappingItem(DocumentSchema):
     # lang => localized string
     value = DictProperty()
 
+    @property
+    def escaped_key(self):
+        """
+        Return the key with spaces replaces by underscores for use in xml
+        variable names
+        """
+        return self.key.replace(" ", "_")
```

Comment:
(This is basically the suggestion that you linked to, except that it makes an effort to use something human-readable if easy, and falls back to a hash if not)